lib/geordi/settings.rb

```diff
@@ -67,47 +67,34 @@ def read_settings
       local_path = LOCAL_SETTINGS_FILE_NAME
       firefox_path = File.expand_path("../../.firefox-version")
 
-      if File.exists?(global_path)
-        global_settings = YAML.safe_load(File.read(global_path)) || {}
-        check_for_invalid_keys(global_settings, ALLOWED_GLOBAL_SETTINGS, global_path)
-      end
-
-      if File.exists?(local_path)
-        local_settings = YAML.safe_load(File.read(local_path)) || {}
-        check_for_invalid_keys(local_settings, ALLOWED_LOCAL_SETTINGS, local_path)
-      end
-
-      if File.exists?(firefox_path)
-        Geordi::Interaction.warn "Unsupported setting \".firefox-version\". Please remove the file in #{firefox_path}."
-        exit 1
-      end
+      global_settings, local_settings = validate_configuration(global_path, local_path, firefox_path)
 
       @global_settings = global_settings || {}
       @local_settings = local_settings || {}
     end
 
+    def validate_configuration(global_path, local_path, firefox_path)
+      return if ENV['GEORDI_INVALID_SETTINGS_WARNED']
+
+      global_settings = File.exists?(global_path) ? YAML.safe_load(File.read(global_path)) || {} : {}
+      check_for_invalid_keys(global_settings, ALLOWED_GLOBAL_SETTINGS, global_path)
+
+      local_settings = File.exists?(local_path) ? YAML.safe_load(File.read(local_path)) || {} : {}
+      check_for_invalid_keys(local_settings, ALLOWED_LOCAL_SETTINGS, local_path)
+
+      Interaction.warn "Unsupported config file \".firefox-version\". Please remove #{firefox_path}." if File.exists?(firefox_path)
+
+      ENV['GEORDI_INVALID_SETTINGS_WARNED'] = 'true'
+      [global_settings, local_settings]
+    end
+
     def check_for_invalid_keys(settings = [], allowed_keys, file)
```

Comment:
Can you remove the default value? Since `settings` usually is a Hash, the Array is misleading. Also, a leading default value behaves unintuitive, as `check_for_invalid_keys([:a], [:b])` would result in arguments `[], [:a], [:b]`.